tests/plugins/filesystem/test_walkfs.py

```diff
@@ -57,3 +58,46 @@ def test_benchmark_walkfs(target_bare: Target, benchmark: BenchmarkFixture) -> N
     result = benchmark(lambda: next(WalkFSPlugin(target_bare).walkfs()))
 
     assert result.path == "/"
+
+
+@pytest.fixture
+def mock_fs_entry() -> MagicMock:
+    """Fixture to create a mock FilesystemEntry object."""
+    mock_entry = MagicMock(spec=FilesystemEntry)
+    mock_fs = Mock()
+    mock_volume = Mock()
+    mock_disk = Mock()
+
+    # Set up the mock object structure
+    mock_entry.fs = mock_fs
+    mock_fs.volume = mock_volume
+    mock_volume.disk = mock_disk
+
+    # Clear any attributes from previous tests
+    mock_volume.guid = None
+    mock_fs.__type__ = "generic"
+    for attr in ("ntfs", "extfs", "fatfs", "exfat"):
+        if hasattr(mock_fs, attr):
+            setattr(mock_fs, attr, None)
+    if hasattr(mock_disk.vs, "serial"):
+        mock_disk.vs = None
+
+    return mock_entry
+
+
+def test_get_disk_serial(mock_fs_entry: MagicMock) -> None:
+    """Test get_disk_serial when a serial number is available."""
+    # Mock the `serial` attribute on the `vs` object
+    mock_fs_entry.fs.volume.vs = Mock(serial="A1B2C3D4")
+
+    assert get_disk_serial(mock_fs_entry.fs) == "A1B2C3D4"
```

Comment:
```suggestion
def test_get_disk_serial() -> None:
    """Test get_disk_serial when a serial number is available."""

    vfs = VirtualFilesystem()

    # Mock the `serial` attribute on the `vs` object
    vfs.volume = Mock(vs=Mock(serial="A1B2C3D4"))

    assert get_disk_serial(vfs) == "A1B2C3D4"
```
This feels a bit cleaner compared to the custom fixture